lib/player.js

```diff
@@ -1596,6 +1596,40 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     });
   }
 
+  /**
+   * Unloads the currently playing stream, if any, and returns a PreloadManager
+   * that contains the loaded manifest of that asset, if any.
+   * Allows for the asset to be re-loaded by this player faster, in the future.
+   * When in src= mode, this unloads but does not make a PreloadManager.
+   *
+   * @param {boolean=} initializeMediaSource
+   * @param {boolean=} keepAdManager
+   * @return {!Promise.<?shaka.media.PreloadManager>}
+   * @export
+   */
+  async unloadAndSavePreload(
+      initializeMediaSource = true, keepAdManager = false) {
+    let preloadManager = null;
+    if (this.manifest_ && this.parser_ && this.parserFactory_ &&
+        this.assetUri_) {
+      // We have enough information to make a PreloadManager!
+      const startTimeOfLoad = Date.now() / 1000;
+      preloadManager = await this.makePreloadManager_(
+          this.assetUri_, /* startTime= */ null, this.mimeType_,
```

Comment:
We should save the currentTime and set it as startTime. For live is more complicate because it has to be relative, but it is possible.

Reply:
Done!